src/NotFoundException.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Http\Discovery;
+
+@trigger_error('The '.__NAMESPACE__.'\NotFoundException class is deprecated since version 1.0 and will be removed in 2.0. Use Http\Discovery\Exception\NotFoundException instead.', E_USER_DEPRECATED);
```

Comment:
I took this namespace thing from symfony. Since the namespace declaration is just above the warning, I think we are allowed to make it shorter in the code. The message itself is what's important after all.

But you are right, maybe we shouldn't throw this as long as we actually throw this exception.